### DISCUSSION TITLE
Can we use go types in Class struct?

### DISCUSSION THREAD

**Discussion by ullanar** [quaadgras/graphics.gd#49] [Q&A] [answered]
Hello! According to discussion https://github.com/grow-graphics/gd/discussions/43 if we making exported field - that field will be exported to Godot. But can I do not export field and just use it internally? 

I mean:
Here is actual struct 

```go
type Model struct {
	classdb.Extension[Model, Node.Instance]
	actions      map[string]actions.Action
	activeAction actions.Action
	parentRoot   CharacterBody2D.Instance
}

```

On Register method in `main.go` we getting panic `panic: gdextension.RegisterClass: unsupported property type string`. I verified what if we will remove `actions  map[string]actions.Action`  field from struct then problem will be resolved

So is it possible to use regular go types in case we just don`t need godot types? That field will never be exported in editor and if it possible - i want to write basic logic on go using go types

**Comment by ullanar** [comment]
Hah

Actually I have found reason

```go
func (m *Model) SwtichAction(actionName string) {
	m.activeAction.Exit()
	m.activeAction = m.actions[actionName]
	m.activeAction.Enter()
}
```

All was cos if that method which accepts string as param. So am I understand correctly - it is impossible to accept params in methods which are godot classes?

**Reply by Splizard** [reply]
You can make the method private, is that not an option here?

**Reply by ullanar** [reply]
yay, didnt thought what would work 

Thanks, yes. Still breaking my mind with unusual lib :)

**Comment by Splizard** [comment, marked as answer]
You can now use Go types in a `classdb.Extension` struct, just make sure they are exported.
